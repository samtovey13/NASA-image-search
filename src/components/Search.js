import '../styles/Search.css';
import React from 'react';

const Search = () => {
  return <> 
    <input 
      className="search-input"
      type="text"
      placeholder="Search"
      data-testid="search-input"
    />
  </>
}

export default Search;