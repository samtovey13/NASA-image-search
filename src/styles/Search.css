.search-form {
  display: flex;
  flex-direction: row;
  align-items: stretch;
}

.search-input, .search-button {
  height: 50px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
  margin: 5px;
  box-sizing: border-box;
}