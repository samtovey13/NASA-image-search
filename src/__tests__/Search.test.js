import { render } from '@testing-library/react';
import Search from '../components/Search';

describe ("Search", () => {
  it ("renders correctly", () => {
    const { asFragment } = render( <Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it ("renders the correct components", () => {
    const { getByTestId, getByPlaceholderText } = render(<Search />);

    expect(getByTestId("search-input")).toHaveClass("search-input");
    expect(getByPlaceholderText("Search")).toBeInTheDocument();
  });
});