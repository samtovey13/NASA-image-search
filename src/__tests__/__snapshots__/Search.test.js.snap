// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <input
    class="search-input"
    data-testid="search-input"
    placeholder="Search"
    type="text"
  />
</DocumentFragment>
`;
